### PR TITLE
Change the warning when creating project with name app

### PR DESCRIPTION
**Pull request by TsvetanMilanov**
When creating project with name app a warning will appear and a prompt to create the project with name app or change the project name.

Implements the functionality described in https://github.com/NativeScript/nativescript-cli/issues/1203#issuecomment-203367786

### CODE REVIEW

File: lib/services/project-name-service.ts [hunk 1]
```diff
@@ -15,16 +15,27 @@ export class ProjectNameService implements IProjectNameService {
...
+				return this.promptForNewName("The project name is invalid.", projectName, validateOptions).wait();
 			}
 
+			let userCanInteract = isInteractive();
+
 			if (!this.checkIfNameStartsWithLetter(projectName)) {
-				if (!isInteractive()) {
+				if (!userCanInteract) {
 					this.$errors.fail("The project name does not start with letter and will fail to build for Android. If You want to create project with this name add --force to the create command.");
 					return;
```
Comment (rosen-vladimirov) [inline lib/services/project-name-service.ts]: let's remove this `return`, it's just confusing for me

File: lib/services/project-name-service.ts [hunk 2]
```diff
@@ -15,16 +15,27 @@ export class ProjectNameService implements IProjectNameService {
...
 				}
 
-				return this.promptForNewName(projectName, validateOptions).wait();
+				return this.promptForNewName("The project name does not start with letter and will fail to build for Android.", projectName, validateOptions).wait();
+			}
+
+			if (projectName.toUpperCase() === "APP") {
+				if (!userCanInteract) {
+					this.$errors.fail("You cannot build applications named 'app' in Xcode. Consider creating a project with different name. If You want to create project with this name add --force to the create command.");
+					return;
```
Comment (rosen-vladimirov) [inline lib/services/project-name-service.ts]: let's remove this `return`, it's just confusing for me

**Comment by rosen-vladimirov** [comment]
:+1:  great work

**Comment by vchimev** [comment]
run ci

**Pull request merged**